### PR TITLE
Description of BackedEnum choices type for Assert\Choice constraint

### PR DESCRIPTION
**Pull request by goodjinny**
Documentation describes solution for the next  [issue](https://github.com/symfony/symfony/issues/44564).

### CODE REVIEW

File: reference/constraints/Choice.rst [hunk 1]
```diff
@@ -321,12 +321,19 @@ to return the choices array. See
 ``choices``
 ~~~~~~~~~~~
 
-**type**: ``array`` [:ref:`default option <validation-default-option>`]
+**type**: ``array|BackedEnum`` [:ref:`default option <validation-default-option>`]
```
Comment (OskarStark) [inline reference/constraints/Choice.rst]:
```suggestion
**type**: ``array`` or ``\BackedEnum`` [:ref:`default option <validation-default-option>`]
```

File: reference/constraints/Choice.rst [hunk 2]
```diff
@@ -321,12 +321,19 @@ to return the choices array. See
...
 A required option (unless `callback`_ is specified) - this is the array
 of options that should be considered in the valid set. The input value
 will be matched against this array.
 
+If `BackedEnum <https://www.php.net/manual/en/language.enumerations.backed.php>`_ is passed,
+all the enum values will be converted to an array and matched the same way.
+
+.. versionadded:: 6.1
+
+    The \BackedEnum choices type was introduced in Symfony 6.1.
```
Comment (OskarStark) [inline reference/constraints/Choice.rst]: This is about Validation, no form types.

The support for BackedEnum was introduced in....

Comment (goodjinny) [reply reference/constraints/Choice.rst]: Fixed. Thank you @OskarStark!

File: reference/constraints/Choice.rst (hunk 1 above)
Comment (goodjinny) [reply reference/constraints/Choice.rst]: Fixed. Thank you!

**Comment by wouterj** [comment]
I see the related code PR was closed, so I'll close this PR as well.

Thanks anyways for immediately providing the docs for suggestion you made on the code, @goodjinny! I would be happy to merge a PR of you in the future :)

**Pull request closed without merging**